occameracontrol/__main__.py

```diff
@@ -60,6 +60,22 @@ def control_camera(camera: Camera):
         time.sleep(1)
 
 
+def update_cameras(cameras: list[Camera]):
+    update_frequency = config_t(int, 'camera_update_frequency') or 60
+    logger.info(str(update_frequency))
+    error_handlers = {
+        camera.url: RequestErrorHandler(
+                camera.url,
+                f'Failed to communicate with camera {camera}')
+        for camera in cameras}
+    while True:
+        for camera in cameras:
+            with error_handlers[camera.url]:
+                camera.move_to_preset(camera.position)
```

Comment:
The problem with this is that it is not free of side effects. If this runs while the other thread changes the camera position, the camera objects does not yet have the correct position, the camera would be turned back, but this tool would think that the camera has the correct position.

Working on the same resource in parallel is always a hassle and can easily lead to mistakes/race conditions. I would avoid that unless you actually need it. In this case, I think we can easily live with just one thread, executing the position reset in the same method where we handle the position changes.

I would probably try adding this to the end of `Camera::update_position`, turning the `if…else` into a `if…elif…else`.

Reply:
Okay I changed this as you said. But now the control_camera function has to sleep for the update frequency, otherwise the cameras are reset every second. Therefore, I changed the default update frequency to 30 in order to not have a too large difference between start / end times and position updates.

Reply:
Can you not keep the 1s loop but do the position reset only every 'n' seconds by e.g. adding something like a `last_updated` property to the camera and then doing a check like `if time.time() - self.last_updated > update_frequency`?